allennlp/modules/stacked_bidirectional_lstm.py

```diff
@@ -104,8 +111,11 @@ def forward(self,  # pylint: disable=arguments-differ
 
             output_sequence = torch.cat([forward_output, backward_output], -1)
             output_sequence = pack_padded_sequence(output_sequence, lengths, batch_first=True)
-            final_states.append((torch.cat(both_direction_states, -1) for both_direction_states
-                                 in zip(final_forward_state, final_backward_state)))
 
-        final_state_tuple = (torch.cat(state_list, 0) for state_list in zip(*final_states))
+            final_h.extend([final_forward_state[0], final_backward_state[0]])
+            final_c.extend([final_forward_state[1], final_backward_state[1]])
```

Comment:
wouldn't the equivalent of the previous code be something like

```
final_h.append(torch.cat([final_forward_state[0], final_backward_state[0]], -1))
```

?

was that wrong? or is this the same and I'm just not seeing it?

Reply:
That is equivalent to the previous code, however that would not work with the Seq2Vec wrapper/interface as this module is Bi-Directional the `final_h` and `final_c` should be of shape *(num_layers * num_directions, batch, hidden_size)* which the code I have put in the pull request will be where as the previous code and your suggestion I believe would be of shape *(num_layers, batch, hidden_size)* the place that this would causes an error is in lines 104 to 109 in [PytorchSeq2VecWrapper](https://github.com/allenai/allennlp/blob/master/allennlp/modules/seq2vec_encoders/pytorch_seq2vec_wrapper.py) as the module has the `bidirectional` attribute therefore it would reshape the `state` incorrectly. Sorry I did not mention this problem in the issue or pull request.

Reply:
I am a little wary about changing the shape of that state, but it doesn't break any tests and it makes the seq2vecencoder work, so 🤞